drivers/MbedCRC.h

```diff
@@ -644,7 +644,7 @@ class MbedCRC {
     }
 
     /* Check to see if we can do assembler optimizations */
-#if ((defined __GNUC__ || defined __clang__) && !defined __CC_ARM) && \
+#if ((defined __GNUC__ || defined __clang__)) && \
```

Comment:
One set of parenthesis can be removed.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/66e537acbb25f3fb5a6aa7b5d844b09ae2fddf02..e435738cd71f66f0ce2787651876f781928a1ac3).